appserver/appclient/client/acc/src/main/java/org/glassfish/appclient/client/acc/ACCClassLoader.java

```diff
@@ -187,27 +181,15 @@ private Class<?> findClassUnshadowed(String name) throws ClassNotFoundException
 
     private byte[] readByteCode(final String className) throws ClassNotFoundException {
         String resourceName = className.replace('.', '/') + ".class";
-        InputStream is = getResourceAsStream(resourceName);
-        if (is == null) {
-            throw new ClassNotFoundException(className);
-        }
-
-        try {
-            ByteArrayOutputStream baos = new ByteArrayOutputStream();
-            byte[] buffer = new byte[8196];
-            int bytesRead;
-            while ((bytesRead = is.read(buffer)) != -1) {
-                baos.write(buffer, 0, bytesRead);
+        try (InputStream is = getResourceAsStream(resourceName)) {
+            if (is == null) {
+                throw new ClassNotFoundException(className);
             }
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            FileUtils.copy(is, baos, is.available());
```

Comment:
3rd argument represents hopefully the buffer size not the size to copy? (this FileUtils doesn't seem to be from Apache Commons so I couldn't verify).

Reply:
FileUtils attracted my attention in my second workspace, here it uses NIO and the third is used as a size.
`        outChannel.transferFrom(inChannel, 0, size);`

>  The maximum number of bytes to be transferred; must be non-negative

Maybe would be better to simply use Long.MAX_VALUE ...?

Reply:
Probably.

This just caught my attention because `is.available()` ranks among the top of beginner's misunderstandings.

Reply:
To summarize: 
* in.available is correct here, because the input is a file and the third argument is a max count of bytes to copy.
* In next PR I will review FileUtils and document these methods, and probably explicitly warn to avoid using Long.MAX_VALUE, and also add loop to be sure we read all we expected, because based on FileChannel javadoc it is not guaranteed to be done in a single step.